subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -424,6 +427,45 @@ public void setDebug(boolean enabled) {
         forkOptions.setDebug(enabled);
     }
 
+    /**
+     * Enables fail fast behavior causing the task will fail on the first failed test.
+     */
+    @Option(option = "fail-fast", description = "Stops test execution after the first failed test.")
+    public Test failFastOn(boolean ignored) {
+        setFailFast(true);
+        return this;
+    }
+
+    /**
+     * Disables fail fast behavior causing the task to run all tests.
+     */
+    @Option(option = "no-fail-fast", description = "Ensures test execution continues after a failed test.")
+    public Test failFastOff(boolean ignored) {
+        setFailFast(false);
+        return this;
+    }
+
+    /**
+     * Sets the 'failFast' property controlling if this task will fail on the first failed test.
+     *
+     * @param failFast boolean whether the task will fail on the first failed test
+     */
+    @Override
+    public void setFailFast(boolean failFast) {
+        super.setFailFast(failFast);
+    }
+
+    /**
+     * Indicates if this task will fail on the first failed test
+     *
+     * @return whether this task will fail on the first failed test
+     */
+    @Input
```

Comment:
❌ This is not an input - it doesn't change the output of a successful execution. Failures are not cached by Gradle.

Reply:
done